### PR TITLE
[astgen] Introduce ASTNode; update ResultType to be ASTNode; use ASTNode to implement BraceStmt correctly.

### PR DESCRIPTION
**Pull request by zoecarver**

### CODE REVIEW

File: include/swift/AST/CASTBridging.h [hunk 1]
```diff
@@ -128,8 +128,13 @@ void *SwiftVarDecl_create(void *ctx, BridgedIdentifier _Nullable name,
...
 
-void *BraceStmt_createExpr(void *ctx, void *lbloc, BridgedArrayRef elements, void *rbloc);
-void *BraceStmt_createStmt(void *ctx, void *lbloc, BridgedArrayRef elements, void *rbloc);
+struct ASTNodeBridged {
+  void *ptr;
+  _Bool isExpr; // Must be expr or stmt.
+};
+
+void *BraceStmt_create(void *ctx, void *lbloc, BridgedArrayRef elements, void *rbloc);
+void *BraceStmt_create(void *ctx, void *lbloc, BridgedArrayRef elements, void *rbloc);
```
Comment (zoecarver) [inline include/swift/AST/CASTBridging.h:137]: I need to get rid of one of these, oops.

**Comment by zoecarver** [comment]
@swift-ci please smoke test.

File: lib/ASTGen/Sources/ASTGen/Decls.swift [hunk 2]
```diff
@@ -4,35 +4,37 @@ import SwiftSyntax
...
+  public func visit(_ node: TypealiasDeclSyntax) -> ASTNode {
     let aliasLoc = self.base.advanced(by: node.typealiasKeyword.position.utf8Offset).raw
     let equalLoc = self.base.advanced(by: node.initializer.equal.position.utf8Offset).raw
     var nameText = node.identifier.text
     let name = nameText.withUTF8 { buf in
       return SwiftASTContext_getIdentifier(ctx, buf.baseAddress, buf.count)
     }
     let nameLoc = self.base.advanced(by: node.identifier.position.utf8Offset).raw
-    let genericParams = node.genericParameterClause.map(self.visit)
+    let genericParams = node.genericParameterClause.map(self.visit).map { $0.rawValue }
```
Comment (AnthonyLatsis) [inline lib/ASTGen/Sources/ASTGen/Decls.swift:15]: `map { self.visit($0).rawValue }`?

Comment (zoecarver) [reply lib/ASTGen/Sources/ASTGen/Decls.swift:15]: Thanks, Anthony!

**Pull request merged**